File: Stack and Queues/Valid Parentheses.cpp
class Solution {
public:
    bool isValid(string str) {
        stack<char> s;
        for(int i = 0; i < str.length(); i++){
            char c = str[i];
            if(c == '{' || c == '[' || c == '('){
                s.push(c);
            }
            else if(s.empty()){
                return false;
            }
            else{
                char t = s.top();
                if(t == '{' && c == '}')
                    s.pop();
                else if(t == '[' && c == ']')
                    s.pop();
                else if(t == '(' && c == ')')
                    s.pop();
                else
                    return false;
            }
        }
        return s.empty();
    }
};

File: Stack and Queues/Valid Parentheses.rs
impl Solution {
    pub fn is_valid(s: String) -> bool {
        let mut stack = vec![];

        for ch in s.chars() {
            match ch {
                '(' => stack.push(')'),
                '{' => stack.push('}'),
                '[' => stack.push(']'),
                ')' | '}' | ']' => {
                    match stack.pop() {
                        Some(x) if x == ch => (),
                        _ => return false,
                    };
                },
                _ => return false,
            }
        }
        stack.is_empty()
    }
}